python/pyarrow/tests/test_compute.py

```diff
@@ -1702,6 +1702,18 @@ def test_logical():
 
 
 def test_cast():
+    arr = pa.array([1, 2, 3, 4], type='int64')
+    options = pc.CastOptions()
+
+    with pytest.raises(ValueError):
+        pc.cast(arr, target_type=None)
+
+    with pytest.raises(ValueError):
+        pc.cast(arr, 'int32', safe=True, options=options)
+
+    with pytest.raises(ValueError):
+        pc.cast(arr, 'int32', safe=False, options=options)
+
```

Comment:
Thanks for testing the error cases, can we also test that the non-error case now works with cast options?

Reply:
I think we should also still test with a non-default `pc.CastOptions`, to ensure we are really testing that the options object is passed through (the test below `pc.cast(arr, 'int8', options=options)` would also pass if `options` was not specified)